core/src/test/scala/unit/kafka/server/BaseAdminRequestTest.scala

```diff
@@ -0,0 +1,102 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.server
+
+import java.io.{DataInputStream, DataOutputStream}
+import java.net.Socket
+
+import kafka.client.ClientUtils
+import kafka.cluster.Broker
+import kafka.integration.KafkaServerTestHarness
+import kafka.network.SocketServer
+import kafka.utils._
+import org.apache.kafka.common.protocol.{Errors, SecurityProtocol}
+import org.junit.Before
+
+/**
+ * TODO: This is a pseudo-temporary test implementation to test admin requests while we still do not have an AdminClient.
```

Comment:
AdminClient.scala is a temporary Scala implementation as part of KIP-40, but it was really a stopgap to get 0.9 out. This AdminClient I am referring too is the java client that will be added as a part of the clients package as the public admin client. I will update to clarify that I mean the java client. cc @hachikuji